server/metadata/tenant.go

```diff
@@ -611,7 +620,24 @@ func (tenant *Tenant) CreateDatabase(ctx context.Context, tx transaction.Tx, dbN
 
 	// otherwise, proceed to create the database if there are concurrent requests on different workers then one of
 	// them will fail with duplicate entry and only one will succeed.
-	_, err := tenant.metaStore.CreateDatabase(ctx, tx, dbName, tenant.namespace.Id())
+	dbId, err := tenant.metaStore.CreateDatabase(ctx, tx, dbName, tenant.namespace.Id())
+	if config.DefaultConfig.Auth.EnableNamespaceIsolation {
```

Comment:
do we need this check here? if accessToken is present then we store it by default?

Reply:
if namespace isolation is enabled, access token must be not nil here.

Reply:
this is no longer valid with metadata being created from outside.